RFS/src/main/java/com/rfs/common/OpenSearchClient.java

```diff
@@ -81,23 +85,19 @@ private Optional<ObjectNode> createObjectIdempotent(String objectPath, ObjectNod
         if (response.code == HttpURLConnection.HTTP_NOT_FOUND) {
             client.put(objectPath, settings.toString());
             return Optional.of(settings);
-        } else if (response.code == HttpURLConnection.HTTP_OK) {
-            logger.info(objectPath + " already exists. Skipping creation.");
-        } else {
-            logger.warn("Could not confirm that " + objectPath + " does not already exist. Skipping creation.");
-        }
+        } 
+        // The only response code that can end up here is HTTP_OK, which means the object already existed
         return Optional.empty();
     }
 
     /*
-     * Attempts to register a snapshot repository; no-op if the repo already exists.  Returns an Optional; if the repo
-     * was created, it will be the settings used and empty if it already existed.
+     * Attempts to register a snapshot repository; no-op if the repo already exists.  
      */
-    public Optional<ObjectNode> registerSnapshotRepo(String repoName, ObjectNode settings){
+    public void registerSnapshotRepo(String repoName, ObjectNode settings){
         String targetPath = "_snapshot/" + repoName;
-        RestClient.Response response = client.putAsync(targetPath, settings.toString())
+        client.putAsync(targetPath, settings.toString())
             .flatMap(resp -> {
-                if (resp.code == HttpURLConnection.HTTP_OK || resp.code == HttpURLConnection.HTTP_CREATED) {
+                if (resp.code == HttpURLConnection.HTTP_OK) {
```

Comment:
Why make this change?

Reply:
Good question.  After testing it more carefully, the API wasn't actually returning 201's regardless of whether we created the snapshot ourselves or found it to be already there.  So in that sense, a return value of 201 would be unexpected.

I don't have a strong opinion here if you'd prefer to keep them both as acceptable responses.